lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +35,64 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self.path_to_data = path_to_raw_txt_data
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self.path_to_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_data.is_dir():
+            raise NotADirectoryError
+
+        if not any(self.path_to_data.iterdir()):
+            raise EmptyDirectoryError
+
+        raw_files = [file for file in self.path_to_data.glob('*_raw.txt')]
+        meta_files = [file for file in self.path_to_data.glob(r'*_meta.json')]
+
+        if len(meta_files) != len(raw_files):
+             raise InconsistentDatasetError
+
+        for file in raw_files:
+            if not file.stat().st_size:
+                raise InconsistentDatasetError
+
+        for file in meta_files:
+            if not file.stat().st_size:
+                raise InconsistentDatasetError
+
+        list_of_raw_ids = []
+        for file in raw_files:
+            list_of_raw_ids.append(int(file.name[:file.name.index('_')]))
+        if sorted(list_of_raw_ids) != list(range(1, len(list_of_raw_ids) + 1)):
+            raise InconsistentDatasetError
+
+        list_of_meta_ids = []
+        for file in raw_files:
```

Comment:
It looks like you ment to iterate over meta files, but instead iterate over raw ones